vtex/toolbelt
# Show upload progress and project size

#### What is the purpose of this pull request?
- Show upload progress
- Show project original size and compressed size
- Transfer Builder client code to toolbelt, to avoid doing a breaking change on `node-vtex-api` 
- Add a 90MB limit to project original size

#### How should this be manually tested?

Install this builder-hub version:
```
vtex install vtex.builder-hub@0.187.0-beta.1
```
Install this toolbelt branch:
```
git clone https://github.com/vtex/toolbelt.git && \
cd toolbelt && \
git checkout feat/link-upload-progress && \
yarn && yarn global add file:$PWD
```

Now link a big app. In linux you can create a big file using:
```
dd if=/dev/urandom of=bigfile bs=1024 count=${_MEGABYTES_}K
```

![Screenshot from 2019-12-11 19-08-56](https://user-images.githubusercontent.com/26463288/70664741-bf8b7700-1c49-11ea-9cbe-ed68d927a794.png)

Should test `vtex test` and `vtex publish` as well

#### Types of changes
